### ISSUE TITLE
Should the template set up logging?

### ISSUE THREAD

**Issue by filiph** [brianegan/new_flutter_template#13]
Logging is one of those things every app will need sooner or later. Should it be set up in the template? To what extent?

* No logging in the template
* A comment in the appropriate place (`main()`) saying something like: "Here's where you can set up your logging and error reporting" (plus link)
* Use `pkg:logging` but do not gather the messages
* Use `pkg:logging` and gather the messages in `main` through `Logger.root.onRecord.listen`, printing them
* Use some other approach

**Comment by fullmers** [comment]
Since most projects will need logging, I think it should be included with a simple example.

**Comment by PiotrWpl** [comment]
I think that it would be a good idea to show that there are better solutions for logging than a simple `print()` command. Even showing that there is something like `log()` will be useful for most of the new users.

**Comment by escamoteur** [comment]
Maybe not only logging but also a global exception handler?

**Comment by chimon2000** [comment]
Perhaps a logging abstraction over print that can be easily overridden?

**Comment by nerder** [comment]
This is totally needed, we are struggling on implement this on a mature app. Logging is that kind of thing that is very easy to plug in in the beginning, not so easy to do at later stage, so up for it. 👍🏼

**Comment by sfshaza2** [comment]
Yes to logging, if I get a vote.  :)

**Comment by asidt** [comment]
Yes to logging, possibly a complete but simple example on how to set it up correctly

**Comment by neiljaywarner** [comment]
Flimber or similar - especially if you can do LineNumberDebugTree - jump to line in source https://stackoverflow.com/questions/38689399/log-method-name-and-line-number-in-timber - it's like magic 

Timber is practically the gold standard on Android

especially nice would be a link to a dead simple crashlytics tutorial and a way you  you can turn on with a few lines
